File: Nkandla_methods/Nkandla/Politician.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Nkandla
{
    public class Politician : Person, Visitor
    {

        public Politician(string name, int age, Gender gender, PersonType type, PartyType politicalParty) : base(name, age, gender, type, politicalParty)
        {
            
        }

        public void visit(HomeStead homeStead)
        {
            if (base.getType().Equals(PersonType.Politician) && !base.getPoliticalParty().Equals(PartyType.ANC))
                Console.WriteLine(base.getName() + ": Pay back the money!!!");
            else
                Console.WriteLine(base.getName() + ", welcome to Nkandla!");
        }

        public string ToString()
        {
            return "Politician" + base.ToString();
        }
    }
}

File: Nkandla_Java/src/main/java/com/bbd/Politician.java
package com.bbd;


/**
 * Created by bbdnet1523 on 2016/12/06.
 */
@SuppressWarnings("DefaultFileTemplate")
public class Politician extends Person implements IVisitor {

    public Politician(String name, int age, Gender gender, PersonType politician) {
        super(name, age, gender, politician);
    }

    @Override
    public String toString() {
        return "Politician{} " + super.toString();
    }

    public void visit(Homestead home) {
        if (super.getType().equals(PersonType.Politician))
            throw new RuntimeException("Pay back the money!");

    }
}
